src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaListener.cs

```diff
@@ -112,9 +116,19 @@ internal class KafkaListener<TKey, TValue> : IListener, IScaleMonitorProvider
             return builder.Build();
         }
 
+        private KafkaMetricsProvider<TKey, TValue> CreateMetricsProvider()
+        {
+            return new KafkaMetricsProvider<TKey, TValue>(this.topicName, new AdminClientConfig(GetConsumerConfiguration()), consumer.Value, this.logger);
+        }
+
         private KafkaTopicScaler<TKey, TValue> CreateTopicScaler()
         {
-            return new KafkaTopicScaler<TKey, TValue>(this.listenerConfiguration.Topic, this.consumerGroup, this.functionId, this.consumer.Value, new AdminClientConfig(GetConsumerConfiguration()), this.listenerConfiguration.LagThreshold, this.logger);
+            return new KafkaTopicScaler<TKey, TValue>(this.listenerConfiguration.Topic, this.consumerGroup, this.functionId, this.consumer.Value, metricsProvider.Value, this.listenerConfiguration.LagThreshold, this.logger);
+        }
+
+        private KafkaTargetScaler<TKey, TValue> CreateTargetScaler()
+        {
+            return new KafkaTargetScaler<TKey, TValue>(this.listenerConfiguration.Topic, this.consumerGroup, this.functionId, this.consumer.Value, metricsProvider.Value, this.listenerConfiguration.LagThreshold, this.logger);
```

Comment:
Same here, do we always want to return a new instance?

Reply:
Similar to the last conversation, to quote it for current variable -
> This is where we're utilizing functionality of a Lazy variable. It is instantiated once through the function CreateTargetScaler() during its first use, and its value is retained until the end of process. Since Listener is alive along with the worker, we're creating an instance only once (during its first use) and returning the same instance every time.